elastalert/elastalert.py

```diff
@@ -715,12 +714,9 @@ def set_starttime(self, rule, endtime):
             if 'minimum_starttime' in rule and rule['minimum_starttime'] > buffer_delta:
                 rule['starttime'] = rule['minimum_starttime']
             # If buffer_time doesn't bring us past the previous endtime, use that instead
-            elif 'previous_endtime' in rule:
-                if rule['previous_endtime'] < buffer_delta:
-                    rule['starttime'] = rule['previous_endtime']
-                    self.adjust_start_time_for_overlapping_agg_query(rule)
-                elif rule.get('allow_buffer_time_overlap'):
```

Comment:
is it safe to drop this functionality?

Reply:
I'm not sure I would call this functionality. `allow_buffer_time_overlap` is just a toggle to fix this bug. It's possible someone has started relying on this bug's behavior, but that seems unlikely.